benchmark/CMakeLists.txt

```diff
@@ -80,6 +80,7 @@ set(SWIFT_BENCH_MODULES
     single-source/DictionaryRemove
     single-source/DictionarySubscriptDefault
     single-source/DictionarySwap
+    single-source/Differentiation
```

Comment:
Should this be gated by the `SWIFT_ENABLE_EXPERIMENTAL_DIFFERENTIABLE_PROGRAMMING` CMake flag?
(It's currently [true by default](https://github.com/apple/swift/blob/948372b87ce977096b1c57ce8a0163f60333bc58/utils/build_swift/build_swift/driver_arguments.py#L1069).)

Reply:
Good point. However, I'll gate it by a `#if canImport(_Differentiation)` surrounding the `Differentiation.swift` file because there are multiple ways to build this (CMake and SwiftPM), and that gate works independently of the build system.